language-id.el

```diff
@@ -135,7 +135,14 @@
     ("Python" python-mode)
     ("R" ess-r-mode (ess-mode (ess-dialect "R")))
     ("Racket" racket-mode)
-    ("Reason" reason-mode)
+    ("Reason"
+     (reason-mode
+      (language-id--file-name-extension ".re")
```

Comment:
Should we omit `(language-id--file-name-extension ".re")` so that `reason-mode` buffers with no filename (e.g. go to the `*scratch*` buffer and do `M-x reason-mode`) are identified as `Reason`?

Reply:
If we do that, then `language-id--mode-match-p` never matches ReScript.

Reply:
It should if we swap the order of `Reason` and `ReScript`. Does that work?

Reply:
Oh good idea, last change does that.

Reply:
Looks perfect, thanks!